apollo-federation/src/sources/connect/header.rs

```diff
@@ -302,20 +281,73 @@ mod tests {
         );
     }
 
-    #[rstest]
-    #[case(JSON::Array(vec!["one".into(), "two".into()]), Ok("[\"one\",\"two\"]".into()))]
-    #[case(JSON::Bool(true), Ok("true".into()))]
-    #[case(JSON::Null, Ok("null".into()))]
-    #[case(JSON::Number(1.into()), Ok("1".into()))]
-    #[case(JSON::Object(Map::new()), Ok("{}".into()))]
-    #[case(JSON::String("string".into()), Ok("string".into()))]
```

Comment:
I have... mixed feeling about `rstest`. Curious about what lead to converting to standard tests in this particular case?

Reply:
I had feedback from the router team on a different PR that they were opposed to using macro expansion during tests as it would drive up compile time. A nice trick is that AI an convert `rstest` into regular tests if you just highlight the `rstest` and say "remove rstest", allowing you to get the declarative benefit of defining test cases with `rstest`, with the compile time efficiency of not using it.